src/org/parosproxy/paros/view/MainFrame.java

```diff
@@ -658,7 +658,8 @@ public void setShowTabNames(boolean showTabNames) {
 	 */
 	public void setTitle(Session session) {
 		StringBuilder strBuilder = new StringBuilder();
-		strBuilder.append(session.getSessionName()).append(" - ");
+		String sessionName = session == null ? "" : session.getSessionName();
+		strBuilder.append(sessionName).append(" - ");
 		if (!session.isNewState()) {

```

Comment:
It also requires the null check here (could have an if wrapping all "session statements").